crates/ruma-events/src/room/message/media.rs

```diff
@@ -0,0 +1,29 @@
+//! Reusable methods for media files.
+
+use crate::room::message::FormattedBody;
+
+/// Computes the filename of a media file based on the spec.
+///
+/// If the `filename` field is set, it is used. Otherwise, the `body` field is used.
+pub(crate) fn filename<'a>(body: &'a str, filename: Option<&'a str>) -> &'a str {
```

Comment:
I am not sure that this method is really useful given the simplicity of the logic.

Reply:
Hint: rust-analyzer had a "inline into all callers" assist (though it might still produce slightly weird-looking code).

Reply:
> I am not sure that this method is really useful given the simplicity of the logic.

Ok can do. Both these PRs feels a bit weird to me coming from Swift where it would all be in a protocol extension 🙈

> Hint: rust-analyzer had a "inline into all callers" assist (though it might still produce slightly weird-looking code).

Doesn't Rust inline things like this automatically?